compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3115,7 +3115,11 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
               case xtree: untpd.NameTree => typedNamed(xtree, pt)
               case xtree => typedUnnamed(xtree)
 
+          val hadErroneus = result.tpe.isErroneous
```

Comment:
```suggestion
          val hadErroneous = result.tpe.isErroneous
```

Reply:
`isErroneous` look like it performs a type traversal, so it seems expensive for no reason in the happy path.

Consider storing `val unsimplifiedType = result.tpe` instead, and only check `unsimplifiedType.isErroneous` later, in the guard of `case e: ErrorType`. This way, we avoid the traversal in the happy path where we don't get an `ErrorType`.